NotationV0/src/install.ts

```diff
@@ -0,0 +1,48 @@
+import * as path from 'path';
+import * as taskLib from 'azure-pipelines-task-lib/task';
+import * as toolLib from 'azure-pipelines-tool-lib/tool';
+
+import { getDownloadInfo, installFromURL } from './lib/install';
+
+import { NOTATION_BINARY } from './lib/constants';
+
+const NOTATION_VERSION_FILE = 'notation_versions.json';
+
+export async function install(): Promise<void> {
+    // the notation download URL
+    let version: string
+    let downloadURL: string
+    let checksum: string
+
+    if (taskLib.getBoolInput('isCustomVersion', false)) {
+        // for custom version, download the notation from specified URL
+        version = "1.0.0-UNRELEASED"
+        downloadURL = taskLib.getInput('url', true) || '';
+        checksum = taskLib.getInput('checksum', true) || '';
+    } else {
+        const versionRange = taskLib.getInput('version', true);
+        if (!versionRange) {
+            throw new Error('Version is not specified');
+        }
+        const downloadInfo = getDownloadInfo(versionRange, NOTATION_VERSION_FILE);
+        version = downloadInfo.version;
+        downloadURL = downloadInfo.url;
+        checksum = downloadInfo.checksum;
+    }
+
+    // install notation binary
+    const extractPath = taskLib.getVariable('Agent.TempDirectory');
+    if (!extractPath) {
+        throw new Error('Agent.TempDirectory is not set');
+    }
+    await installFromURL(downloadURL, checksum, extractPath);
+
+    // add to path for subsequent tasks
+    taskLib.prependPath(extractPath);
+    // add to path for current process
+    process.env['PATH'] = `${extractPath}${path.delimiter}${process.env['PATH']}`;
+    // cache tool
+    toolLib.cacheDir(extractPath, NOTATION_BINARY, version);
```

Comment:
Do we really need to cache the notation binary? (given if we allow users to call `install` more than one time in the pipeline and overwrite the notation cli binary)

Reply:
We need to the cache to store the version of Notation. The process to install plugin needs to know the current notation version.

Reply:
If `sign` process needs the `notation CLI` version, should it get it from the `notation binary` from `path` instead of a cache? The concern here is, in-cache happened in the `install` command. If somehow the cache gets flushed in the pipeline before `sign`, would that be a cache miss in the `sign` process?

Reply:
Godd point. Have removed the azure-kv version lock mechanism and alwasy download the latest v1.x azure-kv to simplify the logic. We don't need cacheDir now.